libs/dh/wholesale/feature-request-calculation/src/lib/dh-wholesale-request-calculation.ts

```diff
@@ -0,0 +1,270 @@
+
+    if (!range) return null;
+
+    const rangeInDays = differenceInDays(parseISO(range.end), parseISO(range.start));
+    if (rangeInDays > 31) {
+      return { maxOneMonthDateRange: true };
+    }
+
+    return null;
+  };
+
+const label = (key: string) => `wholesale.requestCalculation.${key}`;
+
+type FormType = {
+  processType: FormControl<EdiB2CProcessType | null>;
+  period: FormControl<WattRange<string | null>>;
+  gridarea: FormControl<string | null>;
+  meteringPointType: FormControl<MeteringPointType | null>;
+  energySupplierId: FormControl<string | null>;
+  balanceResponsibleId: FormControl<string | null>;
+};
+
+@Component({
+  selector: 'dh-wholesale-request-calculation',
+  templateUrl: './dh-wholesale-request-calculation.html',
+  standalone: true,
+  styles: [
+    `
+      :host {
+        display: flex;
+        justify-content: center;
+
+        watt-card {
+          width: 70%;
+        }
+
+        watt-dropdown,
+        watt-datepicker {
+          width: 50%;
+        }
+      }
+    `,
+  ],
+  imports: [
+    WATT_CARD,
+    WattDropdownComponent,
+    WattButtonComponent,
+    DhDropdownTranslatorDirective,
+    VaterStackComponent,
+    VaterFlexComponent,
+    ReactiveFormsModule,
+    FormsModule,
+    TranslocoDirective,
+    WattDatepickerComponent,
+    WattFieldErrorComponent,
+    JsonPipe,
+    NgIf,
+  ],
+})
+export class DhWholesaleRequestCalculationComponent {
+  private _apollo = inject(Apollo);
+  private _fb = inject(NonNullableFormBuilder);
+  private _transloco = inject(TranslocoService);
+  private _toastService = inject(WattToastService);
+  private _destroyRef = inject(DestroyRef);
+  private _selectedEicFunction: EicFunction | null | undefined = null;
+
+  maxDate = subDays(new Date(), 5);
+
+  form = this._fb.group<FormType>({
+    processType: this._fb.control(null, Validators.required),
+    period: this._fb.control({ start: null, end: null }, [
+      Validators.required,
+      WattRangeValidators.required(),
+      maxOneMonthDateRangeValidator(),
+    ]),
+    energySupplierId: this._fb.control(null, Validators.required),
+    balanceResponsibleId: this._fb.control(null, Validators.required),
+    gridarea: this._fb.control(null, Validators.required),
+    meteringPointType: this._fb.control(null, Validators.required),
+  });
+
+  gridAreaOptions: WattDropdownOptions = [];
+  energySupplierOptions: WattDropdownOptions = [];
+  balanceResponsibleOptions: WattDropdownOptions = [];
+
+  meteringPointOptions = enumToDropdownOptions(MeteringPointType);
+  progressTypeOptions = enumToDropdownOptions(EdiB2CProcessType);
+
+  selectedActorQuery = this._apollo.watchQuery({
+    useInitialLoading: true,
+    notifyOnNetworkStatusChange: true,
+    query: graphql.GetSelectedActorDocument,
+  });
+
+  energySupplierQuery = this._apollo.watchQuery({
+    useInitialLoading: true,
+    notifyOnNetworkStatusChange: true,
+    query: graphql.GetActorsForRequestCalculationDocument,
```

Comment:
suggestion: add mocks to `GetActorsForRequestCalculation` and `getSelectedActor` queries

![image](https://github.com/Energinet-DataHub/greenforce-frontend/assets/1096332/3db71ea1-6fa5-47cb-9f85-5422c7324809)